src/PPN-microbench/memory.cpp

```diff
@@ -0,0 +1,81 @@
+#include <PPN-microbench/memory.hpp>
+
+
+// Function to generate a random index within a given threshold
+inline u64 random_index(u64 threshold) { return rand() % threshold; }
+
+// Function to measure cache/memory latency using pointer chasing
+double measure_latency(u64 size, u64 nbIterations) {
+    std::vector<void *> memblock(size);
```

Comment:
Use a regular array here as well. You're even setting your vector size ahead of time so it's an easy change